test/pdo_sqlsrv/pdo_fetch_complex_transactions.phpt

```diff
@@ -0,0 +1,159 @@
+--TEST--
+Test transactions commit, rollback and aborting in between
+--FILE--
+﻿﻿<?php
+include 'pdo_tools.inc';
+
+function ComplexTransaction($conn, $tableName)
+{
+    $stmt = $conn->query("CREATE TABLE $tableName ([c1_int] int, [c2_real] real)");
+    $stmt = null;
+
+    $stmtSelect = $conn->prepare("SELECT * FROM $tableName");
+    $stmtDelete = $conn->prepare("DELETE TOP(3) FROM $tableName");
+   
+    // insert ten rows
+    $numRows = 10;
+    InsertData($conn, $tableName, $numRows);
+    FetchData($stmtSelect, $tableName, $numRows);
+    
+    $conn->beginTransaction();
+    $stmtDelete->execute(); 
+    $rowsAffected = $stmtDelete->rowCount();
+    $conn->commit();
+    echo "Committed deleting 3 rows\n";
+    
+    $numRows = $numRows - $rowsAffected;
+    FetchData($stmtSelect, $tableName, $numRows);
+    
+    $conn->beginTransaction();
+    $stmtDelete->execute(); 
+    $conn->rollback();
+    echo "Rolled back\n";
+    
+    FetchData($stmtSelect, $tableName, $numRows);
+
+    $conn->beginTransaction();
+    $stmtDelete->execute(); 
+    $rowsAffected = $stmtDelete->rowCount();
+    $conn->commit();
+    echo "Committed deleting 3 rows\n";
+    
+    $numRows = $numRows - $rowsAffected;
+    FetchData($stmtSelect, $tableName, $numRows);
+    
+    $conn->beginTransaction();
+    $stmtDelete->execute(); 
+    $conn->rollback();
+    echo "Rolled back\n";
+    
+    FetchData($stmtSelect, $tableName, $numRows);
+
+    $conn->beginTransaction();
+    $stmtDelete->execute(); 
+
+    echo "Deletion aborted\n";  
+
+    return $numRows;    
+}
+
+function InsertData($conn, $tableName, $count)
+{
+    $stmt = $conn->prepare("INSERT INTO $tableName (c1_int, c2_real) VALUES (?, ?)");
+
+    for ($i = 0; $i < $count; $i++)
+    {
+        $v1 = $i + 1;
+        $v2 = $v1 * 1.5;
+
+        $stmt->bindValue(1, $v1);
+        $stmt->bindValue(2, $v2);
+        $stmt->execute(); 
+    }
+}
+
+function FetchData($stmt, $tableName, $numRows, $fetchMode = false)
```

Comment:
It could be the test is testing to see if $stmt->setFetchMode(PDO::FETCH_LAZY) actually works to change to fetch mode